### PR TITLE
POC: improve navigation menus: add top level tree, sidebars & fix mobile

### PR DESCRIPTION
**Pull request by pimlie**
### ❗ This PR should only be merged after #727 is merged

This is a POC to show that the navigation menu's on the website can be improved quite easily with almost minimal html changes. Please let me know if you like me to work on this further.

- Moved some menu items into dropdowns to ensure the top menu doesn't overflow horizontically
- Moved the table-of-contents into a sidebar
- Improved aria-* attribute usage
- Add styling + javascript to control the menu's on mobile / small screens

Also note that this PR:
- extends https://github.com/GrapheneOS/grapheneos.org/pull/727 (but cannot use that PR as a base for this branch)
- still needs a bit more polishing + edge case fixing, but didn't want to spent more time on it without your approval

### Todo's:
- [x] Fix aria-* attributes in desktop mode (currently implemented using css hover's)
- [x] Check compatibility in chrome / firefox / safari
- [x] Extract javascript into separate file
- [ ] ~Extract icon's into separate files (have used fontawesome svg's atm)~ it's easier to leave the svg's inline so we can style them with css
- [x] Polish & clean css

### Not planned:
- With the new page templates & sidebar toc's it probably makes sense to split some of the large pages into smaller sections and/or restructure the TOC's themselves. But that should be done in a different PR to not mix content & ui changes

#### Desktop:
<details><summary>Default view</summary>

Sidebar
</details>

<details><summary>Scrolled sidebar & dropdown menu open</summary>

image
</details>

#### Mobile:
<details><summary>Default view</summary>

Default
</details>

<details><summary>With main menu open</summary>

With Main menu open
</details>

<details><summary>With main menu & sub menu open</summary>

With main menu & sub menu open
</details>

<details><summary>With TOC open</summary>

With TOC open
</details>

### CODE REVIEW

**Comment by maade93791** [comment]
First of all, thanks for a contribution.
I can't comment much on the HTML code, but the I like the concept.
I would like to mention that it would be easier to review if all the DX related commits will be split out to another pull request.
The following applies to https://github.com/GrapheneOS/grapheneos.org/pull/727 as well.

File: templates/pages/releases.html
Comment (maade93791) [inline templates/pages/releases.html:1]: changes in this file must be made accidentally

**Comment by pimlie** [comment]
@maade69 Thanks, will have a look to split out whats possible.

Just a small note to fix some possible confusion, this PR was branched off the #727 branch. So all commits in this PR except the last one https://github.com/GrapheneOS/grapheneos.org/pull/728/commits/1ebc9dd80670cf7ae06efc8568c78761053cfab5 are identical to #727. I.e. if we would merge #727 then this PR would only have one commit with changes.

Apologies for not explaining that better 🙏

**Comment by pimlie** [comment]
This PR should be good now, will mark this as Ready for review once #727 is merged

**Comment by pimlie** [comment]
Closing this PR as apparently there is no interest into merging these improvements and there are too many conflicts now

**Comment by thestinger** [comment]
There was interest in it but it's very hard to review most of it and it's hard to determine if subjective style and layout changes are done in a way we want.

**Comment by pimlie** [comment]
Thanks, I guess that's always the risk when trying to contribute. Be assured I have no hard feelings about it except for a little disappointment about #727 not being merged cause I emailed to ask how I could help, the work in 727 was suggested and then still it didn't get merged. But again, no hard feelings that's just how it goes sometimes with many competing priorities ;)

**Comment by thestinger** [comment]
The issue was that it was very hard to review, would have needed a fair bit of changes and we never found time for it.

**Pull request closed without merging**